test/test_xpath_parser.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: false
+#
+#  Created by Henrik Mårtensson on 2007-02-18.
+#  Copyright (c) 2007. All rights reserved.
+
+module REXMLTests
+  class TestXPathParser < Test::Unit::TestCase
+    def setup
+      @root_element = _make_service_element(["urn:type1", "urn:type2"], ["http://uri"])
+      @root = @root_element.root
```

Comment:
What is the difference of `@root_element` and `@root`?
In general, root is an XML document but `@root` isn't an XML document, right?

Reply:
In this case there is no document.  This is testing the behavior with a fragment, i.e. just an Element, that has some child elements.  In this case @root looks like a strange UNDEFINED element.  I wasn't sure if I should add a test for that.  It doesn't seem to be relevant to the XPathParser in particular.

Reply:
But also, I've removed `@root`, as it is just confusing, and after removing the redundant tests, it was unsused.